instr.ml

```diff
@@ -254,8 +254,14 @@ let used_vars = function
 
 type 'a dict = (string * 'a) list
 
-type segment = instruction_stream
-type program = segment dict
+type version = label * instruction_stream
+type program = instruction_stream dict
```

Comment:
Is your intent that this is equal to `(version list)`? If so, I'd rather see `version list`, or maybe define `('a, 'b) dict` and use `(label, instruction_stream) dict`.